src/object/schema_manager.c

```diff
@@ -2167,10 +2169,35 @@ sm_downcase_name (const char *name, char *buf, int buf_size)
     }
 
   assert (buf != NULL);
-  assert (intl_identifier_lower_string_size (name) < buf_size);
+  assert (buf_size > 0);
+
+  name_size = intl_identifier_lower_string_size (name);
+  if (name_size >= buf_size || name_size >= DB_MAX_IDENTIFIER_LENGTH)
+    {
+      /* exit on error */
+
+      char name_copy[MAX_PRINT_SIZE] = { '\0' };
+      int n_bytes = (buf_size <= DB_MAX_IDENTIFIER_LENGTH) ? buf_size : DB_MAX_IDENTIFIER_LENGTH;
+
+      if (name_size < MAX_PRINT_SIZE)
+	{
+	  memcpy (name_copy, name, name_size);
+	  name_copy[name_size] = '\0';
+	}
+      else
+	{
+	  memcpy (name_copy, name, (MAX_PRINT_SIZE - 4));
+	  memcpy (name_copy + (MAX_PRINT_SIZE - 4), "...", 4);
+	}
+
+      ERROR_SET_ERROR_2ARGS (error, ER_SM_MAX_LENGTH_EXCEEDED_IDENTIFIER, name_copy, n_bytes);
```

Comment:
Is there a length limit for error messages?
I think it can be solved with just two lines like below.

```
int n_bytes = (buf_size <= DB_MAX_IDENTIFIER_LENGTH) ? buf_size : DB_MAX_IDENTIFIER_LENGTH;
ERROR_SET_ERROR_2ARGS (error, ER_SM_MAX_LENGTH_EXCEEDED_IDENTIFIER, name, n_bytes);
```

Reply:
As you said, I removed this code because the function to check the length was run before.
Thank you.